src/Components/Artwork/SaveButton/useSaveArtworkToLists.ts

```diff
@@ -115,21 +114,41 @@ export const useSaveArtworkToLists = (options: SaveArtworkToListsOptions) => {
       return ResultAction.ShowingAuthDialog
     }
 
-    if (savedListId || artwork.isSavedToCustomLists) {
-      openSelectListsForArtworkModal()
-      return ResultAction.SelectingListsForArtwork
-    }
-
-    const response = await saveToDefaultList()
-    const isSaved = response?.saveArtwork?.artwork?.isSaved
+    if (!artwork.isSavedToAnyList) {
+      // We are saving an unsaved artwork, so we should save it to the default list
+      await saveToDefaultList(false)
+      return ResultAction.SavedToDefaultList
+    } else {
+      // We are unsaving an artwork, so first let's check if it's saved to any custom lists.
+      // If so, we just want to open the modal.
+      // Else, we can unsave it from the default list.
+      const data = await fetchQuery<
+        useSaveArtworkToListsArtworkListInclusionQuery
+      >(
+        relayEnvironment,
+        graphql`
+          query useSaveArtworkToListsArtworkListInclusionQuery(
+            $artworkID: String!
+          ) {
+            artwork(id: $artworkID) {
+              isSavedToList(default: false)
+            }
+          }
+        `,
+        { artworkID: artwork.internalID }
+      ).toPromise()
```

Comment:
This `fetchQuery(…).toPromise()` dance is kind of a new one for me — so this is how we imperatively issue Relay queries outside of the normal component lifecycle, yeah? GTK